pylint/config/option_manager_mixin.py

```diff
@@ -245,25 +244,23 @@ def load_provider_defaults(self):
         for provider in self.options_providers:
             provider.load_defaults()
 
-    def read_config_file(self, config_file=None, verbose=None):
+    def read_config_file(
+        self, config_file: Optional[Path] = None, verbose: bool = False
+    ) -> None:
         """Read the configuration file but do not load it (i.e. dispatching
         values to each option's provider)
         """
-        if config_file is None:
-            config_file = self.config_file
-        if config_file is not None:
-            config_file = os.path.expandvars(os.path.expanduser(config_file))
-            if not os.path.exists(config_file):
-                raise OSError(f"The config file {config_file} doesn't exist!")
-
-        use_config_file = config_file and os.path.exists(config_file)
-        if use_config_file:
+        if config_file:
```

Comment:
```suggestion
        if not config_file:
            if verbose:
                msg = "No config file found, using default configuration"
                print(msg, file=sys.stderr)
            return
       ...
```
Pre-ma-ture re-turn 🎉  This one actually made you buffer 30 lines and 5+ ifs/else that even github hide as it's too long to show

Reply:
I do this in the final version but didn't want to touch the rest of the code too much. It's more about accepting different types of parameters correctly than actually improving this method.

Reply:
Fair enough !